refract-spec.md

```diff
@@ -137,9 +139,490 @@ This is how it would expressed in the compact version.
+
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (string) - A URL
+
+#### Examples
+
+Elements MAY be referenced in remove or local documents.
+
+##### Referencing Remote Element
+
+```json
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "#foo"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. The following rules and constraints apply to namespaces.
+
+1. If there is a conflict, the namespaced element defined or referenced last MUST take precedence
+1. Types defined within the local namespace MUST take precedence over referenced namespaces
+1. Prefixed namespaces are only accessible through using a prefix
+1. Namespaces apply to the element in which they are introduced and all child elements
+1. Namespaces MUST not include the elements nor the data in the elements referenced, but rather include the constraints and conditions around the types defined within that namespace
+
+### Namespaces (Element)
+
+The `namespaces` element is a list of Namespace elements.
+
+#### Properties
+
+- `element` namespaces (string, fixed)
+- `content` (array[Namespace])
+
+### Namespace (enum)
+
+#### Members
+
+- (string) - A resolvable URL to a namespace
+- (object)
+    - `prefix` (string) - A prefix
+    - `href` (string) - An `href` is a resolvable URL to a namespace
+
+#### Example Using Both String and Object Namespace Elements
+
+This example shows a namespace being included by way of a string, and a prefixed namespace.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "namespaces": [
+      "http://example.com/namespace1",
+      { "prefix": "ns2", "href": "http://example.com/namespace2" }
+    ]
+  }
+}
+```
+
+### Prefix (Element)
+
+A prefix is a way to reference elements in prefixed namespaces.
+
+#### Example
+
+This example says the `foo` element is found in the prefixed namespaced prefixed as `ns2`.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "prefix": "ns2"
+  },
+  "content": "baz"
+}
+```
+
+## Extending Elements
+
+Elements by default do not include any of the attributes or content from the element in which they reference. An element MAY be extended in order to inherit from other elements.
+
+### Extend (Element)
+
+The `extend` element provides a way to a way to multiply inherit one or more elements to form a new element. The `extend` element MUST NOT affect the elements being extended, and MUST define a new instance of an element.
+
+The `extend` element MUST do a deep merge of the elements found in the `content`, but MUST NOT include any IDs from the content elements in the final element. Each element found in the `content` MUST derive from the same primitive element type. The elements are merged from first to last.
+
+#### Examples
```

Comment:
The purpose of the extend is purely for declarative and functional reasons. The only thing extend does is take two objects and merge them together. That's it. 

In this PR, elements never inherit from another element. Other elements may be referenced, and elements may provide an ID for the ability to reference, but elements do not inherit properties from one another. 

Extend provides the means by which you may declare that a new element exists by the merging of one or more elements. It does not create a new type (which can be used as the value of `element`), but rather simply a new element.

So there is one place inheritance happens, and then another place where type declaration happens, and that is with the type element. The type element takes one element and provides an ID that may be used as the value of `element`.

Type and Extend may then be used together for multiple inheritance. Multiple elements may be merged together with extend, and then a new type is declared from that extend.

Hope that all makes sense :)